igramarea.cpp

```diff
@@ -2276,6 +2384,61 @@ void IgramArea::writeOutlines(QString fileName){
     ifile.read(buf,32);
     ifile.close();*/
 }
+void IgramArea::writeOutlines(QString fileName){
+
+    if (QFileInfo(fileName).suffix().isEmpty()) { fileName.append(".oln"); }
+
+    QJsonObject j1, jOutline,jInside;
+
+    // write outside outline
+    CircleOutline outside = m_outside;
+    outside.translate(QPointF(cropTotalDx, cropTotalDy));
+    outside.toJson(jOutline);
+    j1["outside_outline"] = jOutline;
+
+    CircleOutline inside = m_center;
+    inside.translate(QPointF(cropTotalDx,cropTotalDy));
+    inside.toJson(jInside);
+    j1["inside_outline"] = jInside;
+
+    QSettings set;
+    double filterRad = set.value("DFT Center Filter",10).toDouble();
+    j1["dft_filter_radius"]=filterRad;
+
+    if (m_edgeMaskWidth != 0) {
+        mirrorDlg &md = *mirrorDlg::get_Instance();
+        j1["edge_mask_width"] = md.aperatureReduction;
+    }
+
+    QJsonArray jRegions;
+    for (int i = 0; i < m_polygons.size(); ++ i){
+        if (m_polygons[i].size() > 0){
+            QJsonArray jPolygon;
+            for (unsigned int j = 0; j < m_polygons[i].size(); ++j){
```

Comment:
QT uses signed types for anything.size(), strange choice. Do not hesitate to use std::size_t for anything from std:: 
Safer than unsigned int.
```suggestion
            for (unsigned int j = 0; j < m_polygons[i].size(); ++j){
```